Repository/AbstractPhpcrRepository.php

```diff
@@ -72,6 +82,58 @@ public function find($query, $language = 'glob')
     }
 
     /**
+     * {@inheritdoc}
+     */
+    public function remove($query, $language = 'glob')
+    {
+        $this->failUnlessGlob($language);
+        $nodes = $this->finder->find($this->resolvePath($query));
+
+        // if the query does not contain a glob pattern, then it applies to an
+        // explicit node and we should throw an exception if it is not found.
+        if (false === $this->globHelper->isGlobbed($query) && 0 === count($nodes)) {
+            throw new ResourceNotFoundException($query);
+        }
+
+        // delegate remove nodes to the implementation
+        $this->removeNodes($nodes);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function move($query, $targetPath, $language = 'glob')
+    {
+        $this->failUnlessGlob($language);
+        $nodes = $this->finder->find($this->resolvePath($query));
+
+        // if the query does not contain a glob pattern, then it applies to an
+        // explicit node and we should throw an exception if it is not found.
+        if (false === $this->globHelper->isGlobbed($query) && 0 === count($nodes)) {
+            throw new ResourceNotFoundException($query);
+        }
+
+        // delegate moving to the implementation
+        $this->moveNodes($nodes, $query, $targetPath);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function clear()
+    {
+        throw new \BadMethodCallException('Clear not supported');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function add($path, $resource)
+    {
+        throw new \BadMethodCallException('Add not supported');
```

Comment:
Removing `add` support for now. It doesn't make much sense to me as we would either:

a. create the payload first via. the implementation, then there is no point in an abstraction.
b. create an empty payload - but that will fail if there are required fields / properties.

Also we do not need this for any of our current use cases, so I think we can safely leave it out.